### PR TITLE
feat(auth-request):added logic for invoke confirm payment NPG method

### PR DESCRIPTION
**Pull request by Fernando-Granato**
#### List of Changes

Added logic to manage the detail field of the authorization request in order to direct the request towards NPG.
Backward compatibility with the previous version of the authorization request has been maintained.
#### Motivation and Context

The change was made to manage the confirm payment to NPG.
#### How Has This Been Tested?
The tests were performed locally

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: env.example [hunk 1]
```diff
@@ -2,4 +2,5 @@ PAGOPA_FUNCTIONS_CHECKOUT_HOST=localhost
 PAGOPA_FUNCTIONS_CHECKOUT_PORT=7071
 ENDPOINT_DELAY=0
 CHECK_STATUS_ADDITIONAL_ATTEMPTS=0
-NPG_APY_KEY=npg-api-key
+NPG_API_KEY=npg-api-key
+PSP_API_KEY=psp-api-key
```
Comment (giovanniberti) [inline env.example:6]: Add this to README :)

File: src/handlers/ecommerce/auth-request.ts [hunk 2]
```diff
@@ -1,16 +1,112 @@
...
+      () => res.status(409).send(error400InvalidRequestBody()),
+      resp => {
+        pipe(
+          resp.details,
+          PaymentInstrumentDetail3.decode,
+          O.fromEither,
+          O.fold(
+            () => {
+              pipe(
+                resp.details,
```
Comment (giovanniberti) [inline src/handlers/ecommerce/auth-request.ts:91]: Given that you use the same error handler (i.e. 409 status + error body) you can refactor this using multiple `O.map`/`O.chain` and then use a single `O.fold` at the end. This way you avoid nesting too many callbacks

File: src/handlers/ecommerce/auth-request.ts [hunk 3]
```diff
@@ -1,16 +1,112 @@
...
+  )();
+};
+
+const processAuthorizationRequest = (req: any, res: any): void => {
+  pipe(
+    req.body,
+    RequestAuthorizationRequest.decode,
+    O.fromEither,
+    O.fold(
+      () => res.status(409).send(error400InvalidRequestBody()),
```
Comment (giovanniberti) [inline src/handlers/ecommerce/auth-request.ts:82]: Status code is 409 but the body is for 400, is it correct?

**Review by ciuffagianluca — APPROVED** [review]
LGTM!

File: env.example (hunk 1 above)
Comment (Fernando-Granato) [reply env.example:6]: Fixed with fc444281c7cb2133873b6f314b1d264b52bcfad6

File: src/handlers/ecommerce/auth-request.ts (hunk 3 above)
Comment (Fernando-Granato) [reply src/handlers/ecommerce/auth-request.ts]: Fixed with f1cee0b54b6c373bbd19fa8e97280b0accdea8ef

File: src/handlers/ecommerce/auth-request.ts (hunk 2 above)
Comment (Fernando-Granato) [reply src/handlers/ecommerce/auth-request.ts]: Fixed with 67b8b2188d6338647d5d5b7135ab2734a5fdc949

**Pull request merged**